hoodie-client/src/test/java/com/uber/hoodie/index/bloom/TestHoodieGlobalBloomIndex.java

```diff
@@ -0,0 +1,356 @@
+        rowChange4);
+
+    JavaRDD<HoodieRecord> recordRDD = jsc.parallelize(Arrays.asList(record1, record2, record3, record5));
+
+    String filename0 = writeParquetFile("2016/04/01", Arrays.asList(record1), schema, null, false);
+    String filename1 = writeParquetFile("2015/03/12", Lists.newArrayList(), schema, null, false);
+    String filename2 = writeParquetFile("2015/03/12", Arrays.asList(record2), schema, null, false);
+    String filename3 = writeParquetFile("2015/03/12", Arrays.asList(record4), schema, null,
+        false);
+
+    // intentionally missed the partition "2015/03/12" to see if the GlobalBloomIndex can pick it up
+    HoodieTableMetaClient metadata = new HoodieTableMetaClient(jsc.hadoopConfiguration(), basePath);
+    HoodieTable table = HoodieTable.getHoodieTable(metadata, config, jsc);
+
+
+    // Add some commits
+    new File(basePath + "/.hoodie").mkdirs();
+
+    // partitions will NOT be respected by this loadInvolvedFiles(...) call
+    JavaRDD<HoodieRecord> taggedRecordRDD = index.tagLocation(recordRDD, jsc, table);
+
+    for (HoodieRecord record : taggedRecordRDD.collect()) {
+      if (record.getRecordKey().equals("000")) {
+        assertTrue(record.getCurrentLocation().getFileId().equals(FSUtils.getFileId(filename0)));
+      } else if (record.getRecordKey().equals("001")) {
+        assertTrue(record.getCurrentLocation().getFileId().equals(FSUtils.getFileId(filename2)));
+      } else if (record.getRecordKey().equals("002")) {
+        assertTrue(!record.isCurrentLocationKnown());
+      } else if (record.getRecordKey().equals("004")) {
+        assertTrue(record.getCurrentLocation().getFileId().equals(FSUtils.getFileId(filename3)));
+      }
+
+    }
+
+  }
+
+  // convert list to map to avoid sorting order dependencies
+  private Map<String, BloomIndexFileInfo> toFileMap(List<Tuple2<String, BloomIndexFileInfo>> filesList) {
+    Map<String, BloomIndexFileInfo> filesMap = new HashMap<>();
+    for (Tuple2<String, BloomIndexFileInfo> t : filesList) {
+      filesMap.put(t._1() + "/" + t._2().getFileName(), t._2());
+    }
+    return filesMap;
+  }
+
+  private String writeParquetFile(String partitionPath, List<HoodieRecord> records, Schema schema, BloomFilter filter,
+                                  boolean createCommitTime) throws IOException, InterruptedException {
+    Thread.sleep(1000);
+    String commitTime = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
+    String fileId = UUID.randomUUID().toString();
+    String filename = FSUtils.makeDataFileName(commitTime, 1, fileId);
+    new File(basePath + "/.hoodie/" + commitTime + ".commit").createNewFile();
+    return writeParquetFile(partitionPath, filename, records, schema, filter, createCommitTime);
+  }
+
+  private String writeParquetFile(String partitionPath, String filename, List<HoodieRecord> records, Schema schema,
```

Comment:
Let's move these methods to HoodieTestUtils. These are reused across TestHoodieBloomIndex..

Reply:
Good point will do.